neuro_evolution_ctrnn/brains/i_brain.py

```diff
@@ -1,8 +1,21 @@
 import abc
-from tools.configurations import IBrainCfg
+from tools.configurations import IBrainCfg, register
 import numpy as np
 from gym.spaces import Space, Discrete, Box, tuple
-from typing import TypeVar, Generic
+from typing import TypeVar, Generic, get_args, Type
+
+registered_brains = {}
+
+
+def register_brain(name: str, config_class: Type[IBrainCfg]):
+    def _register(type_class: type):
+        assert name not in registered_brains, 'type "' + str(name) + '" was already registered'
+        registered_brains[name] = type_class
+        register(name)(config_class)
```

Comment:
```suggestion
        # Registers a brain with the name parameter and then uses the returned _register function to further register the  config class 
        register(name)(config_class)
```

Vielleicht ein Code comment in der Art hier. Hat verwirrt was da genau passiert

Reply:
Dieses Konstrukt `methoden_name(parameter)(Klasse)` verwendet man, wenn einen Decorator nicht als Decorator aufruft, sondern als Methode. Ich habe einen entsprechenden kommentar hinzugefügt